src/lib/bootloader/grub2pwd.rb

```diff
@@ -1,63 +1,131 @@
 require "yast"
 
-# class is responsible for detection, encryption and writing of grub2 password protection
-class GRUB2Pwd
-  YAST_BASH_PATH = Yast::Path.new(".target.bash_output")
-  PWD_ENCRYPTION_FILE = "/etc/grub.d/42_password"
+Yast.import "Stage"
 
-  def used?
-    Yast.import "FileUtils"
+module Bootloader
+  # class is responsible for detection, encryption and writing of grub2 password protection
+  class GRUB2Pwd
+    attr_accessor :used, :unrestricted
+    alias_method :used?, :used
+    alias_method :unrestricted?, :unrestricted
 
-    Yast::FileUtils.Exists PWD_ENCRYPTION_FILE
-  end
+    def initialize
+      # TODO: offline upgrade should somehow read it when we need grub2 -> grub2 modifications
```

Comment:
for grub1 -> grub2 we do repropose, password config is lost and cannot be converted as it use different hashing algorithm